src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                End Select
+            End Function
+
+            Private Function ProcessFromClause(parentExpression As ExpressionSyntax, fromClause As FromClauseSyntax) As ExpressionSyntax
+                Dim expression = parentExpression
+                Dim variables = fromClause.Variables
+                If variables.Count > 1 Then
+                    Return Nothing ' Do not support more than 1 variable. It complicates query methods.
+                End If
+
+                Dim variable = variables.First()
+                If parentExpression Is Nothing Then
+                    expression = variable.Expression
+
+                    If variable.AsClause IsNot Nothing Then
+                        Dim identifier = SyntaxFactory.IdentifierName(NameOf(Enumerable.Cast))
+                        Dim generic = SyntaxFactory.GenericName(
+                            identifier.Identifier,
+                            SyntaxFactory.TypeArgumentList(SyntaxFactory.SingletonSeparatedList(variable.AsClause.Type)))
+                        expression = SyntaxFactory.InvocationExpression(
+                            SyntaxFactory.MemberAccessExpression(
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            expression,
+                            SyntaxFactory.Token(SyntaxKind.DotToken), generic),
+                            SyntaxFactory.ArgumentList())
+                    End If
+
+                    Return expression
+                End If
+
+                Dim anonymousFunctions = FindAnonymousFunctionsFromParentInvocationOperation(variable.Expression)
+                Dim firstLambda = CreateLambdaExpression(anonymousFunctions.First())
+                Dim secondArgumentAnonymousFunction = anonymousFunctions.Last()
+                Dim returnedValue = DirectCast(secondArgumentAnonymousFunction.Body.Operations.First(), IReturnOperation).ReturnedValue
+                Dim secondLambda As LambdaExpressionSyntax
+                If returnedValue.Kind = OperationKind.AnonymousObjectCreation Then
+                    Dim objectCreation = DirectCast(returnedValue, IAnonymousObjectCreationOperation)
+                    Dim fieldInitializers = objectCreation.Initializers.
+                        Select(Function(initializer) SyntaxFactory.InferredFieldInitializer(
+                        SyntaxFactory.IdentifierName(BeautifyName(DirectCast(initializer, IParameterReferenceOperation).Parameter.Name))))
+                    Dim anonymousObjectCreation = SyntaxFactory.AnonymousObjectCreationExpression(
+                        SyntaxFactory.ObjectMemberInitializer(SyntaxFactory.SeparatedList(Of FieldInitializerSyntax)(fieldInitializers)))
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction, anonymousObjectCreation)
+                Else
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction)
```

Comment:
>secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction) [](start = 20, length = 70)

We should make sure we never get here unless we understand where this lambda come from.